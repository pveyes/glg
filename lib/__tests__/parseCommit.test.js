const parseCommit = require("../parseCommit");
const { stripIndent } = require("common-tags");

test("parse commit separated by separator", () => {
  const separator = "####";
  const log = stripIndent`
    87c4fad####fix style parser for encoded quote####Thu Sep 7 14:12:42 2017 +0700####Fri Sep 8 14:02:52 2017 +0700####
  
    M\tlib/style/parser.js
    A\tlib/style/entities.js
  `;

  const commit = parseCommit(log, separator);
  expect(commit).toMatchSnapshot();
});

test("provides `from` for renamed file", () => {
  const separator = "####";
  const log = stripIndent`
    a454fc6####refactor webpack config####Fri Sep 8 22:05:51 2017 +0700####Fri Sep 8 22:08:43 2017 +0700####
  
    R100\tconfig/webpack/desktop.js\tpackages/nodejs-web/config/webpackDesktop.js
    D\twebpack.config.desktop.js
    M\twebpack.config.js
  `;

  const commit = parseCommit(log, separator);
  expect(commit).toMatchSnapshot();
});
