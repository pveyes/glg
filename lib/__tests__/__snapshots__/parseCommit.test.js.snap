// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse commit separated by separator 1`] = `
Object {
  "changes": Array [
    Object {
      "path": "lib/style/parser.js",
      "type": "M",
    },
    Object {
      "path": "lib/style/entities.js",
      "type": "A",
    },
  ],
  "createdDate": "Thu Sep 7 14:12:42 2017 +0700",
  "hash": "87c4fad",
  "publishedDate": "Fri Sep 8 14:02:52 2017 +0700",
  "subject": "fix style parser for encoded quote",
}
`;

exports[`provides \`from\` for renamed file 1`] = `
Object {
  "changes": Array [
    Object {
      "from": "config/webpack/desktop.js",
      "path": "packages/nodejs-web/config/webpackDesktop.js",
      "type": "R100",
    },
    Object {
      "path": "webpack.config.desktop.js",
      "type": "D",
    },
    Object {
      "path": "webpack.config.js",
      "type": "M",
    },
  ],
  "createdDate": "Fri Sep 8 22:05:51 2017 +0700",
  "hash": "a454fc6",
  "publishedDate": "Fri Sep 8 22:08:43 2017 +0700",
  "subject": "refactor webpack config",
}
`;
